File: DSRFQ.Web/Modules/Wallet/WalletTransactions/WalletTransactionsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace DSRFQ.Wallet.Columns;

[ColumnsScript("Wallet.WalletTransactions")]
[BasedOnRow(typeof(WalletTransactionsRow), CheckNames = true)]
public class WalletTransactionsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public int UserId { get; set; }
    public string CompanyName { get; set; }
    [EditLink]
    public string StripeSessionId { get; set; }
    public string StripePaymentIntentId { get; set; }
    public string StripeInvoiceId { get; set; }
    public string StripeInvoiceUrl { get; set; }
    public decimal Amount { get; set; }
    public string Currency { get; set; }
    public string Status { get; set; }
    public string PaymentStatus { get; set; }
    public string CustomerName { get; set; }
    public string CustomerEmail { get; set; }
    public DateTime InsertDate { get; set; }
    public string InsertBy { get; set; }
    public DateTime UpdateDate { get; set; }
    public string UpdateBy { get; set; }
    public DateTime DeleteDate { get; set; }
    public string DeleteBy { get; set; }
}

File: DSRFQ.Web/Modules/ServerTypes/Wallet/WalletTransactionsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { WalletTransactionsRow } from "./WalletTransactionsRow";

export interface WalletTransactionsColumns {
    Id: Column<WalletTransactionsRow>;
    UserId: Column<WalletTransactionsRow>;
    CompanyName: Column<WalletTransactionsRow>;
    StripeSessionId: Column<WalletTransactionsRow>;
    StripePaymentIntentId: Column<WalletTransactionsRow>;
    StripeInvoiceId: Column<WalletTransactionsRow>;
    StripeInvoiceUrl: Column<WalletTransactionsRow>;
    Amount: Column<WalletTransactionsRow>;
    Currency: Column<WalletTransactionsRow>;
    Status: Column<WalletTransactionsRow>;
    PaymentStatus: Column<WalletTransactionsRow>;
    CustomerName: Column<WalletTransactionsRow>;
    CustomerEmail: Column<WalletTransactionsRow>;
    InsertDate: Column<WalletTransactionsRow>;
    InsertBy: Column<WalletTransactionsRow>;
    UpdateDate: Column<WalletTransactionsRow>;
    UpdateBy: Column<WalletTransactionsRow>;
    DeleteDate: Column<WalletTransactionsRow>;
    DeleteBy: Column<WalletTransactionsRow>;
}

export class WalletTransactionsColumns extends ColumnsBase<WalletTransactionsRow> {
    static readonly columnsKey = 'Wallet.WalletTransactions';
    static readonly Fields = fieldsProxy<WalletTransactionsColumns>();
}